cassandra_nodetool/datadog_checks/cassandra_nodetool/cassandra_nodetool.py

```diff
@@ -33,9 +33,22 @@
 class CassandraNodetoolCheck(AgentCheck):
 
     datacenter_name_re = re.compile('^Datacenter: (.*)')
-    node_status_re = re.compile(r'^(?P<status>[UD])[NLJM] +(?P<address>\d+\.\d+\.\d+\.\d+) +'
-                                r'(?P<load>\d+(\.\d*)?) (?P<load_unit>(K|M|G|T)?i?B) +\d+ +'
-                                r'(?P<owns>(\d+(\.\d+)?)|\?)%? +(?P<id>[a-fA-F0-9-]*) +(?P<rack>.*)')
+
+    # nodetool status output
+    node_status_re_pattans = [
+        # --  Address    Load       Tokens  Owns    Host ID                               Rack
+        # UN  127.0.0.1  47.66 KB   1       33.3%   aaa1b7c1-6049-4a08-ad3e-3697a0e30e10  rack1
+        re.compile(r'^(?P<status>[UD])[NLJM] +(?P<address>\d+\.\d+\.\d+\.\d+) +'
+                   r'(?P<load>\d+(\.\d*)?) (?P<load_unit>(K|M|G|T)?i?B) +\d+ +'
+                   r'(?P<owns>(\d+(\.\d+)?)|\?)%? +(?P<id>[a-fA-F0-9-]*) +(?P<rack>.*)'),
+        # old nodetool status output
+        # --  Address    Load       Owns    Host ID                               Token  Rack
+        # UN  127.0.0.1  47.66 KB   33.3%   aaa1b7c1-6049-4a08-ad3e-3697a0e30e10  4035225268091337308       rack1
+        re.compile(r'^(?P<status>[UD])[NLJM] +(?P<address>\d+\.\d+\.\d+\.\d+) +'
+                   r'(?P<load>\d+(\.\d*)?) (?P<load_unit>(K|M|G|T)?i?B) +'
+                   r'(?P<owns>(\d+(\.\d+)?)|\?)%? +(?P<id>[a-fA-F0-9-]*) +-?\d+ +'
```

Comment:
What do you think about making this additional token column part of the original regex by adding this optional bit `(-?\d+ +)?` ? It would avoid the duplication of the rest of the regex which is identical.

Reply:
That is the point I was wondering, and I couldn't make up my mind which is good for us. 
First my PR was prioritize ease of reading over simplicity.

I'll make new commit to use one regex.